pol-core/bscript/compiler/astbuilder/SourceFileProcessor.cpp

```diff
@@ -292,6 +292,42 @@ antlrcpp::Any SourceFileProcessor::visitFunctionDeclaration(
   return antlrcpp::Any();
 }
 
+antlrcpp::Any SourceFileProcessor::visitClassDeclaration(
+    EscriptGrammar::EscriptParser::ClassDeclarationContext* ctx )
+{
+  auto loc = location_for( *ctx );
+  workspace.function_resolver.register_available_class_decl( loc, ctx );
+  auto class_name = tree_builder.text( ctx->IDENTIFIER() );
+
+  // Add var statements to top-level, as their declaration + execution needs to happen in-line with
+  // other statements:
+  //
+  //   Foo::staticVar;  // compilation error
+  //   class Foo var staticVar := 2; endclass
+  //   print(Foo::staticVar == 2);  // true
+  //
```

Comment:
This will change in the scoping PR: accessing `Foo::staticVar` on line 1 will visit the parse tree for `Foo`, adding `staticVar` to its scope.